services/vision/mlvision/detector.go

```diff
@@ -59,17 +59,26 @@ func attemptToBuildDetector(mlm mlmodel.Service) (objectdetection.Detector, erro
 			return nil, err
 		}
 
-		locations := unpack(outMap, "location")
-		if len(locations) == 0 {
-			locations = unpack(outMap, DefaultOutTensorName+"0")
+		locations, err := unpack(outMap, "location")
+		if err != nil || len(locations) == 0 {
+			locations, err = unpack(outMap, DefaultOutTensorName+"0")
+			if err != nil {
+				return nil, err
+			}
 		}
-		categories := unpack(outMap, "category")
-		if len(categories) == 0 {
-			categories = unpack(outMap, DefaultOutTensorName+"1")
+		categories, err := unpack(outMap, "category")
+		if err != nil || len(categories) == 0 {
+			categories, err = unpack(outMap, DefaultOutTensorName+"1")
+			if err != nil {
+				return nil, err
+			}
 		}
-		scores := unpack(outMap, "score")
-		if len(scores) == 0 {
-			scores = unpack(outMap, DefaultOutTensorName+"2")
+		scores, err := unpack(outMap, "score")
+		if err != nil || len(scores) == 0 {
+			scores, err = unpack(outMap, DefaultOutTensorName+"2")
+			if err != nil {
+				return nil, err
```

Comment:
Rather than overwrite `err`, can you combine the two errors? You can use multierr.Combine
https://pkg.go.dev/go.uber.org/multierr#section-readme
you will need to define err2 outside of the scope of the if statement in order for "scores" to be correctly filled, too.